### DISCUSSION TITLE
provideSome is throwing weird errors

### DISCUSSION THREAD

**Discussion by sandeepkota** [zio/zio#6362] [ZIO Contributors]
Right now I'm upgrading zio-k8s to zio 2.0 - https://github.com/coralogix/zio-k8s/pull/310
I have got some weird compilation issue in the `LogsExample` in zio-k8s. As part of upgrade I'm made changes in LogsExample to use .`provideSome` method. Here is the error I'm getting. There is more to it, its a long macros error I guess, I'm upload the text file containing full error. Does anyone know how to fix it? or am I doing anything wrong?

```
[error]   EmptyScope.enter
[error]      while compiling: /Users/sandeepkota/source/zio-ecosystem/zio-k8s/examples/logs-example/src/main/scala/com/coralogix/zio/k8s/examples/logs/LogsExample.scala
[error]         during phase: globalPhase=erasure, enteringPhase=posterasure
[error]      library version: version 2.12.15
[error]     compiler version: version 2.12.15
```
[provideSomeError.txt](https://github.com/zio/zio/files/8060106/provideSomeError.txt)

**Comment by adamgfraser** [comment]
@sandeepkota Thanks for reporting! I believe @kitlangton has made some changes to the macro to avoid these types of errors. @kitlangton Can you confirm?

**Reply by kitlangton** [reply]
Yup! I believe the current series/2.x will fix these issues.

**Reply by sandeepkota** [reply]
Oh that's cool, Thanks :)
